projects/briey/memtest/src/main.c

```diff
@@ -1,26 +1,39 @@
 #include <stdint.h>
+#include <stdlib.h>
 #include <briey.h>
 
-#define PATTERN (0b10101010u)
+#define PATTERN       (0b10101010u)
+
+#define UART_DATA_LEN (8U)
+#define UART_BAUD     (115200U)
 
 extern void flushDataCache(); // From /vga/src/crt.S
 extern int _heap_start, _heap_end;
 volatile uint8_t state = 0;
 
-void fail(){
-    GPIO_A_BASE->OUTPUT = 0xFFFFFFFF ^ 0x04000000 ^ 0x000000FF; // All red LEDs
+void fail(uint32_t errorAddr){
+    GPIO_A_BASE->OUTPUT = (0xFFFFFFFF ^ 0x04000000 ) ^ 0x000000FF; // All red LEDs
+    printf("Error detected at addr %d\r\n", errorAddr);
```

Comment:
Do we need the \r? (Same for everything below)

Reply:
Need it or Putty output will be misaligned: https://superuser.com/questions/654490/putty-new-line-not-working-properly